mathics/eval/arithmetic.py

```diff
@@ -0,0 +1,812 @@
+# -*- coding: utf-8 -*-
+
+"""
+helper functions for arithmetic evaluation, which do not
+depends on the evaluation context. Conversions to Sympy are
+used just as a last resource.
+"""
+
+from typing import List, Optional, Tuple
+
+import mpmath
+import sympy
+
+from mathics.core.atoms import (
+    Complex,
+    Integer,
+    Integer0,
+    Integer1,
+    IntegerM1,
+    Number,
+    Rational,
+    RationalOneHalf,
+    Real,
+)
+from mathics.core.convert.mpmath import from_mpmath
+from mathics.core.convert.sympy import from_sympy
+from mathics.core.element import BaseElement
+from mathics.core.expression import Expression
+from mathics.core.number import dps, min_prec
+from mathics.core.symbols import Atom
+from mathics.core.systemsymbols import (
+    SymbolAbs,
+    SymbolE,
+    SymbolEulerGamma,
+    SymbolI,
+    SymbolKhinchin,
+    SymbolLog,
+    SymbolPi,
+    SymbolPlus,
+    SymbolPower,
+    SymbolTimes,
+)
+from mathics.eval.numbers import eval_accuracy
+
+RationalMOneHalf = Rational(-1, 2)
+RealM0p5 = Real(-0.5)
+RealOne = Real(1.0)
+
+
+# Here we store numerical constants with their approximate
```

Comment:
This smells like a wrong thing to do. I see it is used in `test_arithmetic_expr` so that is probably not advisable either. What is the history behind these?

Reply:
In a couple of places in the code, I need to multiply by that constant. The same with `RationalMOneHalf`. It is just to about avoiding calling the constructor to get the instance.

Reply:
I didn't mean RealOne, I was referring to SymolE, SymbolEulerGamma, and SymbolPi.

Reply:
BTW Computing RealOne once and using it many times is fine. And if we find that other modules tend to need RealOne, we can put it in a more global place.